packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -431,6 +435,29 @@ void paintImage({
     // as we apply a nine-patch stretch.
     assert(sourceSize == inputSize, 'centerSlice was used with a BoxFit that does not guarantee that the image is fully visible.');
   }
+
+  // Output size is fully calculated.
+  assert(() {
+    if (debugImageOverheadAllowedInKilobytes != null) {
+      final int acutalSizeKilobytes = (image.width * image.height * 4 * (4/3)) ~/ 1024;
+      final int possibleSizeKilobytes = (outputSize.width * outputSize.height * 4 * (4/3)) ~/ 1024;
+      final int overhead = acutalSizeKilobytes - possibleSizeKilobytes;
+      if (overhead > debugImageOverheadAllowedInKilobytes) {
+        FlutterError.reportError(FlutterErrorDetails(
+          exception: 'The image $imageTag (${image.width}×${image.height}) '
```

Comment:
make exception a FlutterError object instead of a raw string.

Reply:
Done